### ISSUE TITLE
(cognito): Client Secret handler resource update breaks references

### ISSUE THREAD

**Issue by kylelaker** [aws/aws-cdk#23796]
### Describe the bug

In #23591 `installLatestAwsSdk`. This results in a resource update for custom resources. The custom resource that fetches the client secret does not have an `onUpdate` handler (https://github.com/aws/aws-cdk/blame/0798876e5e6c1a665033c759aed3bc0eab05a892/packages/%40aws-cdk/aws-cognito/lib/user-pool-client.ts#L449).

When the update occurs, the response object does not have a `UserPoolClient.ClientSecret` field, resulting in failures when `.userPoolClientSecret` is referenced.

**This results in stacks that fail to update**

### Expected Behavior

Updates should happen gracefully. The same content as the `onCreate` handler should be run for update events.

### Current Behavior

There is no update handler. Updates fail with an error like:

> CustomResource attribute error: Vendor response doesn't contain UserPoolClient.ClientSecret key in object arn:aws-us-gov:cloudformation:us-gov-west-1:123456789012:stack/StackName/UUID|ResourceId|UUID in S3 bucket cloudformation-custom-resource-storage-usgovwest1

CloudWatch Logs shows:

```
{
    "RequestType": "Update",
    "ResourceType": "Custom::DescribeCognitoUserPoolClient",
    "ResourceProperties": {
        "ServiceToken": "...",
        "InstallLatestAwsSdk": "true",
        "Create": "{\"region\":\"us-gov-west-1\",\"service\":\"CognitoIdentityServiceProvider\",\"action\":\"describeUserPoolClient\",\"parameters\":{\"UserPoolId\":\"...\",\"ClientId\":\"...\"},\"physicalResourceId\":{\"id\":\"...\"}}"
    },
    "OldResourceProperties": {
        "ServiceToken": "...",
        "InstallLatestAwsSdk": "false",
        "Create": "{\"region\":\"us-gov-west-1\",\"service\":\"CognitoIdentityServiceProvider\",\"action\":\"describeUserPoolClient\",\"parameters\":{\"UserPoolId\":\"...\",\"ClientId\":\"...\"},\"physicalResourceId\":{\"id\":\"...\"}}"
    }
}

```

```
{
    "Status": "SUCCESS",
    "Reason": "OK",
    "PhysicalResourceId": "...",
    "StackId": "...",
    "RequestId": "...",
    "LogicalResourceId": "...",
    "NoEcho": false,
    "Data": {}
}

```

I can confirm, but do not want to share, the contents of the `Create` object match between the two events.

### Reproduction Steps

- use a version of the CDK _prior_ to #23591

```ts
import * as cdk from "aws-cdk-lib";
import * as cognito from "aws-cdk-lib/aws-cognito";
import * as secretsmanager from "aws-cdk-lib/aws-secretsmanager";

const app = new cdk.App();
const stack = new cdk.Stack(app, 'TestStack');
const userPool = new cognito.UserPool(stack, 'UserPool');
const client = userPool.addClient('TestClient', { generateSecret: true });
const secret = new secretsmanager.Secret(stack, 'ClientSecret', {
  secretStringValue: client.userPoolClientSecret;
});
```

- update the stack using v2.61.1

### Possible Solution

Copy the `onCreate` handler to `onUpdate`.

### Additional Information/Context

Typically, if the User Pool ID or the User Pool Client ID changes, the User Pool Client Secret would too. But other properties passed to the resource can change (like `installLatestAwsSdk`) and this will result in a resource update

### CDK CLI Version

2.61.1

### Node.js Version

16

### OS

Linux

### Language

Typescript

### Other information

I am preparing a patch that implements the suggested fix.

**Comment by Mattias-** [comment]
Thanks for the report, I'm seeing the same issue. From the changelog it seems like this could be related to the Cognito changes introduced in 2.60.0 https://github.com/aws/aws-cdk/issues/23796

**Comment by kylelaker** [comment]
@Mattias- It was a changed introduced in 2.60.0; however, it was actually the "aws-custom-resource: switch off installLatestAwsSdk by default" fix instead of the Cognito one. At least for the specific issue above.

The new feature that relates to client secrets is for configuring the Google IdP and specifying a Secrets Manager secret for its `clientSecretValue`.

But I do worry _other_ custom resources may have similar issues due to #23591. It'd be probably any custom resource that specifies `onCreate` but not `onUpdate`, had `installLatestAwsSdk` changed to `false`, and has a reference to its return value. So I think that would include:
- `AcceleratorSecurityGroupPeer.fromVpc`: https://github.com/aws/aws-cdk/blob/ba14612b44fb7ccebc7a2b89dd8795e968fcd1c9/packages/%40aws-cdk/aws-globalaccelerator/lib/_accelerator-security-group.ts#L31-L60
- `Table.addPartitionIndex`: https://github.com/aws/aws-cdk/blob/ba14612b44fb7ccebc7a2b89dd8795e968fcd1c9/packages/%40aws-cdk/aws-glue/lib/table.ts#L369-L390
- `EmrContainerStartJobRun.updateRoleTrustPolicy`: https://github.com/aws/aws-cdk/blob/ba14612b44fb7ccebc7a2b89dd8795e968fcd1c9/packages/%40aws-cdk/aws-stepfunctions-tasks/lib/emrcontainers/start-job-run.ts#L330-L345

Also anyone else who uses _any_ `AwsCustomResource`, directly or indirectly would be impacted if they change the feature flag of their existing app and that definition doesn't specify `onUpdate`.

**Comment by joelgarrett** [comment]
This seems to still be broken in 2.62.x and 2.63.x versions.

**Comment by alexb-uk** [comment]
For anyone else that ends up here, I can confirm that this is still an issue for `AcceleratorSecurityGroupPeer.fromVpc`:
```
CustomResource attribute error: Vendor response doesn't contain SecurityGroups.0.GroupId key in object arn:aws:cloudformation:eu-west-1:xxxxx:stack/test/aa-bb-cc|AcceleratorListenerAcceleratorGroupGlobalAcceleratorSGCustomResource0000|cc-bb-aa in S3 bucket cloudformation-custom-resource-storage-euwest1
```
when running the latest CDK `2.75.0`

**Comment by denyskublytskyi** [comment]
Same error with the latest CDK `2.75.1`

`CustomResource attribute error: Vendor response doesn't contain UserPoolClient.ClientSecret key in object`

**Comment by markusz** [comment]
Fixed in `2.79.1`
